google/cloud/aiplatform/metadata/schema/google_schema.py

```diff
@@ -0,0 +1,267 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Optional, Dict
+from google.cloud.aiplatform.metadata.schema import base_artifact
+from google.cloud.aiplatform.metadata.schema import utils
+from google.cloud.aiplatform.compat.types import artifact as gca_artifact
+
+
+class VertexDataset(base_artifact.BaseArtifactSchema):
+    """An artifact representing a Vertex Dataset."""
+
+    SCHEMA_TITLE = "google.VertexDataset"
+
+    def __init__(
+        self,
+        dataset_name: Optional[str] = None,
+        uri: Optional[str] = None,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        state: gca_artifact.Artifact.State = gca_artifact.Artifact.State.LIVE,
+        **kwargs,
+    ):
+        """Args:
+        dataset_name (str):
+            Optional. The name of the Dataset resource, in a form of
+            projects/{project}/locations/{location}/datasets/{datasets_name}. For
+            more details, see
+            https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.datasets/get
+        uri (str):
+            Optional. The uniform resource identifier of the artifact file. May be empty if there is no actual
+            artifact file.
+        display_name (str):
+            Optional. The user-defined name of the Artifact.
+        schema_version (str):
+            Optional. schema_version specifies the version used by the Artifact.
+            If not set, defaults to use the latest version.
+        description (str):
+            Optional. Describes the purpose of the Artifact to be created.
+        metadata (Dict):
+            Optional. Contains the metadata information that will be stored in the Artifact.
+        state (google.cloud.gapic.types.Artifact.State):
+            Optional. The state of this Artifact. This is a
+            property of the Artifact, and does not imply or
+            capture any ongoing process. This property is
+            managed by clients (such as Vertex AI
+            Pipelines), and the system does not prescribe or
+            check the validity of state transitions.
+        **kwargs:
+            Optional. Additional Args that will be passed directly to the Artifact base method for backward compatibility.
```

Comment:
To confirm we do this for the following. Hypothetical example:

```
#v0.0.1
dataset = VertexDataset(..., n_records=10)

# we iterate and remove n_records property
#v0.0.2
dataset = VertexDataset(..., n_records=10)
# still works but doesn't include n_records
```

If this is the case then we should enforce all arguments to be keyword arguments. ie `__init__(self *, ...)` 

Also, if this is the case, I'm concerned about this being interpreted as a breaking change because the same code is executed but my data no longer has the same representation.

Reply:
yes that was my original thought on this arg. however I agree that this will cause a breaking change.  I think to your point the kwargs does not serve the purpose that I had intended for it. I'm removing the kwarg arg.

Reply:
done